tests/integration_tests/bugs/test_lp1835584.py

```diff
@@ -0,0 +1,88 @@
+""" Integration test for LP #1835584
+
+Kernels prior to 4.15 report dmi product_uuid as uppercase, Azure
+relies on system UUID as their "instance-id". Upgrades to kernels >= 4.15
+will represent the same product_uuid as lowercase. System UUID should be a
+case-insensitive comparison and avoid triggering "new instance" detection
+in cloud-init on Azure platform.
+
+The test will launch an Ubuntu image which has a 5.4 kernel. We allow
+cloud-init to detect the instance-id on the 5.4 kernel, set a 4.15 kernel as
+default and reboot the system. Expectation is that cloud-init retains same
+instance-id status and avoids triggering ssh host key regeneration across that
+reboot.
+
+Across the reboot, assert that we didn't re-run config_ssh by virtue of
+seeing only one semaphore creation log entry of type:
+
+ Writing to /var/lib/cloud/instances/<UUID>/sem/config_ssh -
+
+https://bugs.launchpad.net/cloud-init/+bug/1897099
+"""
+import re
+
+import pytest
+
+from tests.integration_tests.instances import IntegrationInstance
+
+
+KERNEL_URL = "https://kernel.ubuntu.com/~kernel-ppa/mainline/v4.14.120/"
+PKG_VER = "4.14.120-0414120-generic_4.14.120-0414120.201905161610_amd64"
```

Comment:
It's my understanding that the generic kernel won't boot on Azure in all cases, and this will presumably only become more true over time. Do we need to find a way of installing an old Azure kernel instead?